packages/@aws-cdk/aws-kinesis/lib/stream.ts

```diff
@@ -746,6 +754,15 @@ export class Stream extends StreamBase {
     });
 
     const shardCount = props.shardCount || 1;
+
+    const streamMode = props.streamModeDetails;
+    let streamModeDetails: CfnStream.StreamModeDetailsProperty | undefined = undefined;
+    if (streamMode !== undefined) {
+      streamModeDetails = {
+        streamMode: StreamModeDetails[streamMode],
+      };
+    }
```

Comment:
We can probably simplify this to (warning: I didn't test this):

```ts
const mode = props.streamModeDetails;

this.stream = new CfnStream(this, 'Resource', {
  ...
  streamModeDetails: mode ? { streamMode: StreamModeDetails[mode] } : undefined;
}
```

Reply:
I'll give this a try after letting the build run with the renames.

Reply:
Loks like it works. Thanks for the suggestion.